File: app/src/main/java/com/ivanovsky/passnotes/data/keepass/KeepassDatabaseKey.java
package com.ivanovsky.passnotes.data.keepass;

import com.ivanovsky.passnotes.data.safedb.EncryptedDatabaseKey;

public class KeepassDatabaseKey implements EncryptedDatabaseKey {

	private final String password;

	public KeepassDatabaseKey(String password) {
		this.password = password;
	}

	@Override
	public byte[] getKey() {
		return password.getBytes();
	}
}

File: app/src/main/java/com/ivanovsky/passnotes/data/repository/keepass/KeepassDatabaseKey.kt
package com.ivanovsky.passnotes.data.repository.keepass

import com.ivanovsky.passnotes.data.repository.encdb.EncryptedDatabaseKey

data class KeepassDatabaseKey(
    private val password: String
) : EncryptedDatabaseKey {

    override val key: ByteArray
        get() = password.toByteArray()
}